docs: UI improve no dataset page to transform it into an onboarding page for first time users developers (#4398)
recognai/rubrix: docs/_source/_common/snippets/start_page.md

Welcome to Argilla! 👋😁 We're thrilled to have you on board. Beginning with Argilla is as simple as it gets, and you've successfully taken the first step!

To start using Argilla, you need to create your first dataset. Let's get started. 🚀

[For in-depth user guides, check the documentation 📖.](https://docs.argilla.io/en/latest/)

> # 1. Open an IDE, Jupyter or Collab

Reviewer: I would try to use h2 (##) instead of h1 (let's see what Leire says but I'll start with that=

> Welcome to Argilla! 👋😁 We're thrilled to have you on board. Starting with Argilla is as simple as it gets, and you've successfully taken the first step! Let's set things in motion. 🚀

Reviewer: ```suggestion
👋😁 Welcome to Argilla!  We're thrilled to have you on board.

To get started you need to create your first dataset using the Python SDK.

Let's see how!
```

> Argilla, as a data curation platform, supports various datasets including `FeedbackDataset`, `DatasetForTextClassification`, `DatasetForTokenClassification`, and `DatasetForText2Text`. So, you should [select the one that best fits your needs](https://docs.argilla.io/en/latest/practical_guides/choose_dataset.html). Note that the FeedbackDataset, adaptable for various NLP tasks, will replace the other more specialized but less versatile datasets in version 2.0.

Reviewer 2: This text is not always as smooth to read for me. Maybe rewrite it a bit.

> Argilla is a data curation platform and supports different types of datasets: the `FeedbackDataset`, which stands out for its versatility across various NLP tasks, and other datasets like the `DatasetForTextClassification`, the `DatasetForTokenClassification` and the `DatasetForText2Text`, which are more feature-rich datasets, but specialized only for these tasks. So, you should [select the one that best fits your needs](https://docs.argilla.io/en/latest/practical_guides/choose_dataset.html). Note that the `FeedbackDataset` will replace the other datasets in version 2.0.

Reviewer: remove this part. We want to direct new users to the Feedback Dataset

Once you've selected a dataset, you can open one of our introductory tutorials to create a [FeedbackDataset](https://colab.research.google.com/github/argilla-io/argilla/blob/develop/docs/_source/getting_started/quickstart_workflow_feedback.ipynb) or [Other datasets](https://colab.research.google.com/github/argilla-io/argilla/blob/develop/docs/_source/getting_started/quickstart_workflow.ipynb). First, you'll need to connect to Argilla as instructed and then just follow the guided process. You'll be able to create a dataset in a few minutes. 🤩

For a more in-depth understanding of Argilla, you can check our documentation 📖:

> * ⚙️ Tweak [server configuration and user management](https://docs.argilla.io/en/latest/getting_started/installation/configurations/configurations.html)

Reviewer: Remove this part. We can include a last sentence point to the docs but not a detailed overview

> * 🗺️ Understand usage workflows using [practical guides](https://docs.argilla.io/en/latest/practical_guides/index.html)

Reviewer: same

> * ✏️ Applied end2end examples using [tutorials](https://docs.argilla.io/en/latest/tutorials_and_integrations/tutorials/tutorials.html)

Reviewer: same

> * 🖥️ Get familiar with the [Argilla UI](https://docs.argilla.io/en/latest/reference/webapp/index.html)

Reviewer: same

Of course, we welcome any ideas or suggestions you may have. And, your contributions will be included in the latest version of Argilla. For more details, take a look at the [contributor guide](https://docs.argilla.io/en/latest/contributing/contributing.html) and the [developer guide](https://docs.argilla.io/en/latest/community/developer_docs.html).

> We hope you enjoy using Argilla as much as we enjoyed building it. If you have any questions, don't hesitate to contact us at the [Argilla Slack community](https://join.slack.com/t/rubrixworkspace/shared_invite/zt-whigkyjn-a3IUJLD7gDbTZ0rKlvcJ5g). We'd love to hear from you! 🙌

Reviewer: Too much text.

> Welcome to Argilla! 👋😁 We're thrilled to have you on board. Beginning with Argilla is as simple as it gets, and you've successfully taken the first step!

Reviewer 2: Maybe "Welcome to Argilla!👋😁 " could be a header and the rest a sub-title?

Author: In fact, there was a header before, but  it got removed

Reviewer: it was me 😆 it felt too much as a docs page, happy to iterate a bit once we see how it renders, could you share a screenshot Sara?

Reviewer 2: Yes I agree, maybe we can ask @leiyre to have a closer look at this?

To start using Argilla, you need to create your first dataset. Let's get started. 🚀

[For in-depth user guides, check the documentation 📖.](https://docs.argilla.io/en/latest/)

# 1. Open an IDE, Jupyter or Collab

Open your favorite IDE, Jupyter or Collab. If you're a Collab user, you can directly use our [introductory tutorial](https://colab.research.google.com/github/argilla-io/argilla/blob/develop/docs/_source/getting_started/quickstart_workflow_feedback.ipynb).

# 2. Install the SDK with pip

To upload and read datasets from Argilla, you need to use the Argilla SDK. You can install the SDK with pip as follows:

```sh

> pip install argilla

Reviewer 2: maybe add an -U to ensure getting the latest version?

```

# 3. Connect to your Argilla server

1. Get your `API_URL`:

   * If you are using Docker, it is the URL shown in your browser (by default `http://localhost:6900`)
   * If you are using HF Spaces, it should be constructed as follows: `https://[your-owner-name]-[your_space_name].hf.space`

> 2. Get your `API_KEY` by clicking on your user icon in the top right corner of the Argilla UI, selecting "My settings" and copying the API key.

Reviewer 2: Did you manage to set this up via a relative URL to `/settings`?

> 3. Connect the SDK with the server using `rg.init`. Make sure to replace "<your-api-url>" and "<your-api-key>" with the values you obtained in the previous steps.

Reviewer 2: Ideally, re-use `ARGILLA_API_KEY`and `ARGILLA_API_URL` as denoted in the docs, or `API_URL`/`API_KEY` or `<your-api-url>`and `<your-api-key>`. Not a mix of everything ;)

```python
import argilla as rg

rg.init(
    api_url="<your-api-url>",
    api_key="<your-api-key>"
)
```

# 4. Create your first dataset

1. Specify a workspace where the dataset is to be saved. You can check your workspaces in the Argilla UI by clicking on "My settings".

> 2. Create a FeedbackDataset with two labels ("sadness" and "joy"). Don't forget to replace "<your-workspace>" with the workspace you want to use:

Reviewer 2: Maybe reference the general `create_dataset`page?

Reviewer 2: And in the reference I would mention that Argilla is fully configurable to your needs but that we will be using one of our templates.

```python
dataset = rg.FeedbackDataset.for_text_classification(
    labels=['sadness', 'joy'],
    multi_label=False,
    use_markdown=True,
    guidelines=None,
    metadata_properties=None,
    vectors_settings=None,
)
dataset.push_to_argilla(name="my-first-dataset", workspace="<your-workspace>")
```

# 5. Add records

Create a list with the records you want to add. Ensure that you match the fields with the ones specified in the previous step.

```python
records = [
    rg.FeedbackRecord(
        fields={
            "text": "I am so happy today",
        },
    ),
    rg.FeedbackRecord(
        fields={
            "text": "I feel sad today",
        },
    )
]
dataset.add_records(records)
```

> You can also use `pandas` or `load_dataset` to [read an existing dataset and create records like above](https://docs.argilla.io/en/latest/practical_guides/create_update_dataset/records.html#add-records).

Reviewer 2: I would maybe move this up to the general get records intro

Welcome to Argilla! Argilla is a platform to build high-quality datasets.

To get started, you can create your first dataset as follows.

[For in-depth user guides, check the documentation 📖.](https://docs.argilla.io/en/latest/)

**1. Open an IDE, Jupyter or Collab**

If you're a Collab user, you can directly use our [introductory tutorial](https://colab.research.google.com/github/argilla-io/argilla/blob/develop/docs/_source/getting_started/quickstart_workflow_feedback.ipynb).

**2. Install the SDK with pip**

To work with Argilla datasets, you need to use the Argilla SDK. You can install the SDK with pip as follows:

```sh
pip install argilla -U
```

**3. Connect to your Argilla server**

Get your `ARGILLA_API_URL`:

   * If you are using Docker, it is the URL shown in your browser (by default `http://localhost:6900`)
   * If you are using HF Spaces, it should be constructed as follows: `https://[your-owner-name]-[your_space_name].hf.space`

> Get your `ARGILLA_API_KEY` you find in ["My settings"](/user_settings) and copy the API key.

Reviewer 3: ```suggestion
Get your `ARGILLA_API_KEY` you find in ["My settings"](/user-settings) and copy the API key.
```

> # Getting started with Argilla

Reviewer: Remove this

Welcome to Argilla! 👋😁 We're thrilled to have you on board. Starting with Argilla is as simple as it gets, and you've successfully taken the first step!

> To start annotating your data, you need to create a `FeedbackDataset`. Let's set things in motion. 🚀

Reviewer: ```suggestion
To start using Argilla, you need to create your first dataset. Let's get started 🚀
```

> [For a more in-depth understanding of Argilla, you can check our documentation 📖.](https://docs.argilla.io/en/latest/)

Reviewer: ```suggestion
[For in-depth user guides,  check the documentation 📖.](https://docs.argilla.io/en/latest/)
```

## 1. Open an IDE, Jupyter or Collab

> 1. Open your favorite IDE, Jupyter or Collab. You can also open our [introductory tutorial](https://colab.research.google.com/github/argilla-io/argilla/blob/develop/docs/_source/getting_started/quickstart_workflow_feedback.ipynb).

Reviewer: ```suggestion
1. Open your favorite IDE, Jupyter or Collab. If you're a Collab user, you can directly use our [introductory tutorial](https://colab.research.google.com/github/argilla-io/argilla/blob/develop/docs/_source/getting_started/quickstart_workflow_feedback.ipynb).
```

> 2. Get your `API_URL`:

Reviewer: This and the next step should be moved to the connect to the Argilla server

   * If you are using Docker, it is the URL shown in your browser (by default `http://localhost:6900`)
   * If you are using HF Spaces, it should be constructed as follows: `https://[your-owner-name]-[your_space_name].hf.space`

3. Get your `API_KEY` by clicking on your user icon in the top right corner of the Argilla UI, selecting "My settings" and copying the API key.

## 2. Install the SDK with pip

> 

Reviewer: To upload and read datasets from Argilla, you need to use the Argilla SDK. You can install the SDK with pip as follows:

```sh
pip install argilla
```

## 3. Connect to your Argilla server

Connect the SDK with the server using `rg.init`. Make sure to replace "<your-api-url>" and "<your-api-key>" with the values you obtained in the first step.

```python
import argilla as rg

rg.init(
    api_url="<your-api-url>",
    api_key="<your-api-key>"
)
```

## 4. Create your first dataset

1. Specify a workspace where the dataset is to be saved. You can check the available workspaces with the following snippet:

```python
workspaces = rg.Workspace.list()
workspace_names = [workspace.name for workspace in workspaces]

> workspace_names

Reviewer: Remove this

```

2. Create a FeedbackDataset with two labels ("sadness" and "joy"). Don't forget to replace "<your-workspace>" with the workspace you want to use:

```python
dataset = rg.FeedbackDataset.for_text_classification(
    labels=['sadness', 'joy'],
    multi_label=False,
    use_markdown=True,
    guidelines=None,
    metadata_properties=None,
    vectors_settings=None,
)
dataset.push_to_argilla(name="my-first-dataset", workspace="<your-workspace>")
```

## 5. Add records

Create a list with the records you want to add. Ensure that you match the fields with the ones specified in the previous step.

```python
records = [
    rg.FeedbackRecord(
        fields={
            "text": "I am so happy today",
        },
    ),
    rg.FeedbackRecord(
        fields={
            "text": "I feel sad today",
        },
    )
]
dataset.add_records(records)
```

> You can also use `pandas` or `load_dataset` to read an existing dataset and create records like above.

Reviewer: Add a link to the most useful section to understand how to add records to a dataset

)
```

## 4. Create your first dataset
# 4. Create your first dataset

> 1. Specify a workspace where the dataset is to be saved. You can check the available workspaces with the following snippet:

Reviewer: well done!

> 1. Specify a workspace where the dataset is to be saved. You can check your workspaces in the Argilla UI by clicking on "My settings".

Reviewer: Maybe add a brief explanation and snippet at the bottom of this section to show how to create a workspace in case they don't have any workspace.

Add a link to the most relevant page for users to read how workspaces work in Argilla

Author: but as far as I could check if they do not remove it, it's created at least one by default (admin in HF and argilla in docker), so would the snippet still be needed?

Reviewer: that's for the quickstart only? maybe @davidberenstein1957 can confirm, but I think you are right, let's not add more code snippets. I'd suggest adding a sentence like: If you want to create a new workspace, check the docs (check the docs is a link with how to create the workspace with python)

Reviewer 2: Yes I believe normally the workspace for the owner is added by default but I would reference the user/workspace setting as Dani suggestion. And I would use the `rg.set_workspace` command after the `rg.init` because I think it is required now.

> 2. Create a FeedbackDataset with two labels ("sadness" and "joy"). Don't forget to replace "<your-workspace>" with the workspace you want to use:

Reviewer: ```suggestion
2. Create a Dataset with two labels ("sadness" and "joy"). Don't forget to replace "<your-workspace>" with the workspace you want to use:
```
